adapters/dynamoDB_adapters/src/main/java/org/atomhopper/dynamodb/query/DynamoDBTextArray.java

```diff
@@ -0,0 +1,198 @@
+package org.atomhopper.dynamodb.query;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.Map;
+
+/**
+ * This is class provides {@link java.sql.Array} interface for PostgreSQL
+ * <code>text</code> array.
+ *
+ *
+ */
+public class DynamoDBTextArray implements java.sql.Array {
+
+    private final String[] stringArray;
+    private final String stringValue;
+
+    /**
+     * Initializing constructor
+     *
+     * @param stringArray
+     */
+    public DynamoDBTextArray(String[] stringArray) {
+        if (stringArray == null) {
+            this.stringArray = null;
+        } else {
+            this.stringArray = Arrays.copyOf(stringArray, stringArray.length);
+        }
+        this.stringValue = stringArrayToPostgreSQLTextArray(this.stringArray);
+    }
+
+    @Override
+    public String toString() {
+        return stringValue;
+    }
+    private static final String NULL = "NULL";
+
+    /**
+     * This static method can be used to convert an string array to string
+     * representation of PostgreSQL text array.
+     *
+     * @param stringArray source String array
+     * @return string representation of a given text array
+     */
+    public static String stringArrayToPostgreSQLTextArray(String[] stringArray) {
```

Comment:
Are we using this particular method? If yes,  the cyclomatic complexity of this method is beyond imagination.